pandas/core/algorithms.py

```diff
@@ -932,7 +932,10 @@ def value_counts_internal(
             idx = Index(keys)
             if idx.dtype == bool and keys.dtype == object:
                 idx = idx.astype(object)
-            elif idx.dtype != keys.dtype:
+            elif (
+                idx.dtype != keys.dtype  # noqa: PLR1714  # # pylint: disable=R1714
```

Comment:
Just confirming this change isn't related to your referenced, closed PR right?

Reply:
No this is fine, this is the infer_string inference, which will still happen